tests/test_selector.py

```diff
@@ -33,6 +33,20 @@ def test_simple_selection(self):
         self.assertEqual([x.extract() for x in sel.xpath("concat(//input[@name='a']/@value, //input[@name='b']/@value)")],
                          [u'12'])
 
+    def test_simple_selection_with_variables(self):
+        """Using XPath variables"""
+        body = u"<p><input name='a' value='1'/><input name='b' value='2'/></p>"
+        sel = self.sscls(text=body)
+
+        self.assertEqual([x.extract() for x in sel.xpath("//input[@value=$number]/@name", number=1)],
+                         [u'a'])
+        self.assertEqual([x.extract() for x in sel.xpath("//input[@name=$letter]/@value", letter='b')],
+                         [u'2'])
```

Comment:
An advantage of variables over string formatting is that you can embed value and not care about escaping, right? Could you please add a test for the case when a variable can break XPath if inserted naively as a string?